src/cli.rs

```diff
@@ -710,68 +681,91 @@ pub fn run(opt: Varlociraptor) -> Result<()> {
+                            let mut processor =
+                                calling::variants::preprocessing::ObservationProcessor::builder()
+                                    .alignment_properties(alignment_properties)
+                                    .protocol_strandedness(protocol_strandedness)
+                                    .max_depth(max_depth)
+                                    .inbam(bam)
+                                    .min_bam_refetch_distance(min_bam_refetch_distance)
+                                    .reference_buffer(Arc::clone(&reference_buffer))
+                                    .breakend_index(BreakendIndex::new(&candidates)?)
+                                    .inbcf(candidates)
+                                    .options(opt_clone)
+                                    .outbcf(output)
+                                    .raw_observation_output(output_raw_observations)
+                                    .log_each_record(log_each_record)
+                                    .realigner(realignment::PathHMMRealigner::new(
+                                        gap_params,
+                                        realignment_window,
+                                        reference_buffer,
+                                    ))
+                                    .build();
+                            processor.process()?;
+                        }
+                        "exact" => {
+                            let mut processor =
+                                calling::variants::preprocessing::ObservationProcessor::builder()
+                                    .alignment_properties(alignment_properties)
+                                    .protocol_strandedness(protocol_strandedness)
+                                    .max_depth(max_depth)
+                                    .inbam(bam)
+                                    .min_bam_refetch_distance(min_bam_refetch_distance)
+                                    .reference_buffer(Arc::clone(&reference_buffer))
+                                    .breakend_index(BreakendIndex::new(&candidates)?)
+                                    .inbcf(candidates)
+                                    .options(opt_clone)
+                                    .outbcf(output)
+                                    .raw_observation_output(output_raw_observations)
+                                    .log_each_record(log_each_record)
+                                    .realigner(realignment::PairHMMRealigner::new(
+                                        reference_buffer,
+                                        gap_params,
+                                        realignment_window,
+                                    ))
+                                    .build();
+                            processor.process()?;
+                        }
+                        _ => panic!("Unknown pairhmm mode '{}'", pairhmm_mode),
+                    };
```

Comment:
Looks like we should now first init the processor, and only add the mode specific args in the match statement.

Reply:
Yes, I tried simplifying that but had some type issues (which aren't even easily solved by box-ing), so I left it as is for the time being.